runners/flink/src/main/java/org/apache/beam/runners/flink/translation/wrappers/streaming/stableinput/BufferingDoFnRunner.java

```diff
@@ -164,6 +171,8 @@ public <KeyT> void onTimer(
       Instant timestamp,
       Instant outputTimestamp,
       TimeDomain timeDomain) {
+
+    currentOutputWatermarkHold = Math.min(timestamp.getMillis(), currentOutputWatermarkHold);
```

Comment:
Shouldn't this be `outputTimestamp` since `outputTimestamp` is the minimum timestamp that is used for producing output for a timer?

```suggestion
    currentOutputWatermarkHold = Math.min(outputTimestamp.getMillis(), currentOutputWatermarkHold);
```

Reply:
Yes, I fixed the independently. Unfortunately, there is some other buf with timers in stable do fn. 
See https://github.com/apache/beam/issues/24662